File: go/0001_TwoSum.go
// Problem 1: Two Sum
// https://leetcode.com/problems/two-sum/

package Go

//lint:file-ignore U1000 Ignore all unused code
func twoSum(nums []int, target int) []int {
	n := len(nums)
	// 宣告回傳的型態
	var output []int

	// 使用雙層迴圈, 第二層的起點為 i+1 避免重複
	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			// 相加等於 target 即找到, 離開迴圈
			if nums[i]+nums[j] == target {
				output = []int{i, j}
				break
			}
		}
	}
	return output
}

File: typescript/0001_TwoSum.ts
// Problem 1: Two Sum
// https://leetcode.com/problems/two-sum/

function twoSum(nums: number[], target: number): number[] {
  // 使用迴圈單層迴圈
  for (let i = 0; i < nums.length; i++) {
    // 如果 希望的值(目標減去當前數值) 在這個array中 且 不是同一個位置的值(index不相同)
    // 就視為找到答案, 直接return
    const complement = target - nums[i];
    const complementIndex = nums.indexOf(complement);

    if (complementIndex !== -1 && i !== complementIndex) {
      return [i, complementIndex];
    }
  }
  return [];
}